web/src/drafts.ts

```diff
@@ -20,45 +21,71 @@ import * as timerender from "./timerender";
 import * as ui_util from "./ui_util";
 import * as util from "./util";
 
-export function set_count(count) {
+export function set_count(count: number): void {
     const $drafts_li = $(".top_left_drafts");
     ui_util.update_unread_count_in_dom($drafts_li, count);
 }
 
-export const draft_model = (function () {
-    const exports = {};
+type LocalStorageDraft = {
+    content: string;
+    updatedAt?: number;
+} & (
+    | {
+          type: false;
+      }
+    | {
+          type: "stream";
+          topic: string;
+          stream_id?: number;
+      }
+    | {
+          type: "private";
+          reply_to: string;
+          private_message_recipient: string;
+      }
+);
+
+// The id is added to the draft in format_drafts in drafts_overlay_ui.
+// We should probably just include it in the draft object itself always?
+type LocalStorageDraftWithId = LocalStorageDraft & {id: string};
 
+export const draft_model = (function () {
     // the key that the drafts are stored under.
     const KEY = "drafts";
     const ls = localstorage();
+
+    // TODO(evy) -- this is causing a type error
+    // and I don't see it used anywhere, can we remove it?
     ls.version = 1;
 
-    function getTimestamp() {
+    function getTimestamp(): number {
         return Date.now();
     }
 
-    function get() {
-        return ls.get(KEY) || {};
+    function get(): Record<string, LocalStorageDraft> {
+        // TODO(evy): Type '{}' is not assignable to type 'Record<string, LocalStorageDraft>'
+        // I don't understand why this is happening. It seems like the problem might
+        // actually be that `ls.get` has unknown return type? What should I do with that?
+        return ls.get(KEY) ?? {};
```

Comment:
Question here!

Reply:
The type for `get` defined in `localstorage.ts` is `unknown`. 

We probably can refactor `localstorage.ts` to have us define something with a type variable such that we're declaring a `ls = new LocalStorage<string, LocalStorageDraft>("drafts")` -- where the key prefix gets backed into the `localstorage` object we get back. Might be a nontrivial refactor given a dozen modules use this, so I'd be tempted to open an issue for that idea and just do a type assertion here.

Reply:
⭐ This is still on my todo list

Reply:
✅